### PR TITLE
KAZOO-3214 crossbar api to access recordings

### PR DESCRIPTION
**Pull request by lazedo**
for @onnet

### CODE REVIEW

File: applications/crossbar/src/modules/cb_recordings.erl [hunk 1]
```diff
@@ -0,0 +1,189 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2012-2016, 2600Hz INC
+%%% @doc
+%%%
+%%% Provides access to stored call recordings:
+%%%
+%%% To play in browser:
+%%% https://my_crossbar.tld:8443/v1/accounts/33ca...1452/recordings/201412-6471-8c79ae@192.1.1.2/attachment?inline=true&auth_token=05c1...7a075
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_recordings.erl]: Can you make these comments ref `/v2` instead?

File: applications/crossbar/src/modules/cb_recordings.erl [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+%% Given the path tokens related to this module, what HTTP methods are
+%% going to be responded to.
+%% @end
+%%--------------------------------------------------------------------
+-spec allowed_methods() -> http_methods().
+-spec allowed_methods(path_token()) -> http_methods().
+-spec allowed_methods(path_token(), path_token()) -> http_methods().
+
+allowed_methods() -> [?HTTP_GET].
+allowed_methods(_Id ) -> [?HTTP_GET].
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_recordings.erl]: Can you name these `_RecordingId`? The auto-gen tool will look nicer with `{RECORDING_ID}` in the path that just `{ID}`.

**Comment by jamesaimonetti** [comment]
@onnet if you have time to test this branch, would love to get your blessing that it works for you. If you don't think you'll get to it, please let us know so we don't hold up the merge if our own testing is successful. Thanks!

**Comment by jamesaimonetti** [comment]
@lazedo for fun, want to run `cb_api_endpoints:to_ref_doc()` and `cb_api_endpoints:to_swagger_json()` to auto-gen the docs for this new endpoint? :) Then copy the ref doc from `crossbar/doc/ref` to `crossbar/doc` and fill in details as appropriate. I'm happy to correct any English mistakes I see ;)

**Comment by onnet** [comment]
@lazedo @jamesaimonetti Will check it for sure. Don't know if I can finish it today (already evening), but tomorrow it will be the main task. But please don't take me into account onto merging it on your own success. If something will be wrong I will spoil your life with tickets or PRs anyway :)
Thanks again!

**Comment by onnet** [comment]
@lazedo @jamesaimonetti Trying to compile KAZOO-3214 with esl-erlang-18.2-1.x86_64.rpm but recieving an error. What release do I need to choose building erlang by kerl?
I am still on CentOS 6

**Comment by jamesaimonetti** [comment]
@onnet 18+ so 18.2 should be fine. What error did you receive when compiling?

**Comment by onnet** [comment]
Yep, Looked at error closer after asking question :) Sorry. But now I know how to use kerl at least...
I just stumbled at building amqp_client:
make[2]: Entering directory `/opt/kazoo/deps/amqp_client'
/bin/sh: -c: line 0: syntax error near unexpected token `('
/bin/sh: -c: line 0: `echo " DEP   " rabbit_common; 	fetch_url1='https://github.com/rabbitmq/rabbitmq-common.git'; fetch_url2='https://github.com/rabbitmq/rabbitmq-common.git'; if test "$fetch_url1" != 'https://github.com/rabbitmq/rabbitmq-erlang-client.git' && git clone -q -n -- "$fetch_url1" /opt/kazoo/deps/rabbit_common; then fetch_url="$fetch_url1"; push_url='https://github.com/rabbitmq/rabbitmq-common.git'; elif git clone -q -n -- "$fetch_url2" /opt/kazoo/deps/rabbit_common; then fetch_url="$fetch_url2"; push_url='git@github.com:rabbitmq/rabbitmq-common.git'; fi; cd /opt/kazoo/deps/rabbit_common && (  git checkout -q (no >/dev/null 2>&1 ||   git checkout -q branch) >/dev/null 2>&1 ||   git checkout -q master >/dev/null 2>&1 ||   git checkout -q master >/dev/null 2>&1 ||  (echo "error: no valid pathspec among:  (no branch) master master" 1>&2 && false) ) && (test "$fetch_url" = "$push_url" || git remote set-url --push origin "$push_url")'
make[2]: Leaving directory `/opt/kazoo/deps/amqp_client'

**Comment by onnet** [comment]
Installed deps manually and have the next error now:
make[4]: *** No rule to make target `/opt/kazoo/deps/rabbitmq_codegen/amqp_codegen.py', needed by `include/rabbit_framing.hrl'.  Stop.

Feels lIke I just miss something. Will try to google

**Comment by jamesaimonetti** [comment]
@onnet is this a fresh clone? Sometimes things get weird if you go from 3.22 back to master. I usually `rm -rf deps/ core/`, then `git checkout`, then `make` from the root to be sure.

**Comment by onnet** [comment]
@jamesaimonetti Nope, it was fresh clone. Will try Debian. Unfortunately it is needed anyway..

**Comment by fenollp** [comment]
@onnet It probably is an `sh` v `bash` issue. What is your `$SHELL`?

**Comment by onnet** [comment]
@fenollp Luis helped with it. He probably will add a couple of strings avoid it in the future.

**Comment by onnet** [comment]
No Debian so far! Hurry!!!! :)

**Comment by onnet** [comment]
@lazedo @jamesaimonetti PR is tested and works like charm for me. 
Many thanks for help in moving my sandbox to master to @lazedo 
Saved loads of time for me!

**Pull request merged**